x/rollapp/types/params.go

```diff
@@ -54,13 +77,36 @@ func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
 		paramtypes.NewParamSetPair(KeyDisputePeriodInBlocks, &p.DisputePeriodInBlocks, validateDisputePeriodInBlocks),
 		paramtypes.NewParamSetPair(KeyDeployerWhitelist, &p.DeployerWhitelist, validateDeployerWhitelist),
 		paramtypes.NewParamSetPair(KeyRollappsEnabled, &p.RollappsEnabled, func(_ interface{}) error { return nil }),
+		paramtypes.NewParamSetPair(KeyLivenessSlashBlocks, &p.LivenessSlashBlocks, validateLivenessSlashBlocks),
+		paramtypes.NewParamSetPair(KeyLivenessSlashInterval, &p.LivenessSlashInterval, validateLivenessSlashInterval),
+		paramtypes.NewParamSetPair(KeyLivenessJailBlocks, &p.LivenessJailBlocks, validateLivenessJailBlocks),
 	}
 }
 
+func (p Params) WithDeployerWhitelist(l []DeployerParams) Params {
+	p.DeployerWhitelist = l
+	return p
+}
+
+func (p Params) WithDisputePeriodInBlocks(x uint64) Params {
+	p.DisputePeriodInBlocks = x
+	return p
+}
+
 // Validate validates the set of params
 func (p Params) Validate() error {
 	if err := validateDisputePeriodInBlocks(p.DisputePeriodInBlocks); err != nil {
-		return err
+		return errorsmod.Wrap(err, "dispute period")
+	}
+
+	if err := validateLivenessSlashBlocks(p.LivenessSlashBlocks); err != nil {
+		return errorsmod.Wrap(err, "liveness slash blocks")
+	}
+	if err := validateLivenessSlashInterval(p.LivenessSlashInterval); err != nil {
+		return errorsmod.Wrap(err, "liveness slash interval")
+	}
+	if err := validateLivenessJailBlocks(p.LivenessJailBlocks); err != nil {
+		return errorsmod.Wrap(err, "liveness jail blocks")
```

Comment:
please add UTs

Reply:
did you have something in mind sorry? I don't think we have any UTs for param validation at the moment anyway
Anyway I have extracted this specific validation to sdk-utils now

Reply:
you can add the test cases to `TestGenesisState_Validate` in `genesis_test.go`